### PR TITLE
[Backport to 15] Translate readnone attribute as function parameter attribute (#1697)

### PR DESCRIPTION
**Pull request by stanleygambarin**
Community restricted readnone, readonly and writeonly attributes
to be only function parameter attributes. This patch aligns
the translator with llvm.org.

It also fixes a bug, when readnone attribute is being mapped
to NoWrite SPIR-V function parameter attribute.

Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

Signed-off-by: Sidorov, Dmitry <dmitry.sidorov@intel.com>

### CODE REVIEW

**Comment by svenvh** [comment]
> Community restricted readnone, readonly and writeonly attributes
> to be only function parameter attributes. This patch aligns
> the translator with llvm.org.

This restriction only applies to LLVM 16 I believe, so not sure if this patch should be cherry-picked in its entirety to older branches?

**Comment by stanleygambarin** [comment]
@MrSidims: i will follow up with b95227e980fd632e478c1faa2fc1544d42b4c355 cherry-pick (https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1746), but can you elaborate on why SPIRVFunctionControlMaskKind should not be changed (and which older LLVM branches) ?

**Comment by MrSidims** [comment]
> @MrSidims: i will follow up with [b95227e](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/b95227e980fd632e478c1faa2fc1544d42b4c355) cherry-pick (#1746), but can you elaborate on why SPIRVFunctionControlMaskKind should not be changed (and which older LLVM branches) ?

Because in LLVM 15 and earlies it's not restricted to have these 2 attributes as function attributes. So the mapping should remain.

**Review by MrSidims — APPROVED** [review]
LGTM thanks! Will merge after @svenvh approves

**Review by svenvh — APPROVED** [review]

**Pull request merged**
